File: Greedy Algorithm/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        volume = 0
        while (left < right):
            volume1 = min(height[left], height[right]) * (right-left)
            volume = max(volume, volume1)
            if(height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return volume
            
        
        
        
                
        

File: 11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int length=height.size();
        int i=0,j=length-1;
        int res=0;
        while(i<j){
            int w=(j-i)*min(height[i],height[j]);
            res=max(res,w);
            if(height[i]<height[j])
                ++i;
            else
                --j;
            //cout<<"i"<<i<<"j"<<j<<"     ";
        }
        return res;
    }
};


